include/laser_filters/radius_outlier_filter.h

```diff
@@ -0,0 +1,190 @@
+    dyn_server_.reset(new dynamic_reconfigure::Server<laser_filters::RadiusOutlierFilterConfig>(own_mutex_, private_nh));
+    dynamic_reconfigure::Server<laser_filters::RadiusOutlierFilterConfig>::CallbackType f;
+    f = boost::bind(&laser_filters::RadiusOutlierFilter::reconfigureCB, this, _1, _2);
+    dyn_server_->setCallback(f);
+
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("radius_search"), radius_search_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given radius_search.\n");
+      return false;
+    }
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("min_neighbors"), min_neighbors_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given min_neighbors.\n");
+      return false;
+    }
+    if (!filters::FilterBase<sensor_msgs::LaserScan>::getParam(std::string("window"), window_))
+    {
+      ROS_ERROR("Error: RadiusOutlierFilter was not given window.\n");
+      return false;
+    }
+
+    if (radius_search_ < 0)
+    {
+      ROS_ERROR("radius_search must be >= 0. Forcing radius_search = 0.");
+      radius_search_ = 0.0;
+    }
+
+    if (min_neighbors_ < 0)
+    {
+      ROS_ERROR("min_neighbors must be >= 0. Forcing min_neighbors = 0");
+      min_neighbors_ = 0;
+    }
+    radius_detector_.configure(radius_search_);
+
+    param_config.radius_search = radius_search_;
+    param_config.window = window_;
+    param_config.min_neighbors = min_neighbors_;
+    dyn_server_->updateConfig(param_config);
+
+    return true;
+  }
+
+  void reconfigureCB(RadiusOutlierFilterConfig& config, uint32_t level)
+  {
+    radius_search_ = config.radius_search;
+    window_ = config.window;
+    min_neighbors_ = config.min_neighbors;
+
+    radius_detector_.configure(radius_search_);
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  virtual ~RadiusOutlierFilter()
+  {
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////
+  /** \brief Filter points based on 3 global parameters: radius_search, min_neighbors and
+   * window. The radius_search specifies the maximum allowed distance for
+   * points to be neighbors. The min_neighbors specifies how many neighboring points
+   * have to be found (within radius_search being the maximum allowed distance) to consider
+   * the point to be valid. The window parameter specifies how many consecutive measurements
+   * to take into account for one point.
+   * \param scan_in the input LaserScan message
+   * \param scan_out the output LaserScan message
+   */
+  bool update(const sensor_msgs::LaserScan& scan_in, sensor_msgs::LaserScan& scan_out)
+  {
+    boost::recursive_mutex::scoped_lock lock(own_mutex_);
+    // copy across all data first
+    scan_out = scan_in;
+
+    std::set<int> indices_to_delete;
+    // For each point in the current line scan
+    for (unsigned int i = 0; i < scan_in.ranges.size(); i++)
```

Comment:
In cases like this where an index variable is going to be used in math expressions, I prefer to just make it an int, and change the upper limit to `i < static_cast<int>(scan_in.ranges.size())`. Not a required change though, since it looks like you cast it correctly down below.

Reply:
Thanks for the note - you're right. I think it's also better to change to C++-style casts in general.
I've done so in c06b709b14027d1bde7be75ed7dc2e16a8f2ece5.